README.md

```diff
@@ -37,6 +37,9 @@ dev.run-ecommerce:   Devstack - Generate microsite configuration in eCommerce
 `config/config.yaml` file contains following features -
 
 - `main_domain`: The base domain. Each microsite will be set as subdomain of this domain.
+- `oauth`: OAuth applications in LMS
+    1. `ecommerce_sso_client`: eCommerce SSO Client name, defaults to `custom-sites-ecommerce-sso`. It is intentionally a different one than the one found by default in an Open edX installation, since the default one resets on every deployment. But we need to have redirect URIs for dynamic custom sites. Generator will create this client autometically if not found.
+    2. `ecommerce_backend_service_client`: eCommerce backend worker client name. Defaults to `ecommerce-backend-service`, which is available by default in an Open edX installation. Generator will not create this client autometically.
```

Comment:
@shimulch, why is this parameter to set the ecommerce backend client even needed? SSO will not use this application at all.

Reply:
@lgp171188 We will be using this to get backend service credentials to add [here](https://github.com/nelc/configuration-secure/blob/master/microsite-config/config.yaml#L98).

Reply:
@shimulch, why do we need a separate OAuth2 application for the backend service app, though?

Reply:
@lgp171188 Those two are using two different OAuth2 flows. SSO using `Authorization code` flow and background service using `Client credentials`. And every service has a separate application for SSO and background workers. For example - 

- edx_notes_api-sso & edx_notes_api-backend-service
- credentials-sso & credentials-backend-service
- discovery-sso & discovery-backend-service

Reply:
@shimulch, the question I have is - why do we need a separate OAuth2 application for the non-SSO purposes? The default applications should suffice, right?

Reply:
@lgp171188 I am not so sure about it. Since, [Client Credentials](https://oauth.net/2/grant-types/client-credentials/#:~:text=The%20Client%20Credentials%20grant%20type,to%20access%20a%20user's%20resources.) (used by background service) flow can have a token outside of user's context and `Authorization Code` flow is about doing work on User's behalf, there can be some issues with having the same application for both use cases. 

And this is something Open edX deployment does by default, having separate applications for SSO and background services. To actually answer whether the same app will work or not, we will have to try it and test it properly.

Reply:
@shimulch, it will. Because it has worked okay till now. The `*-sso` client is a bit special because it uses a browser-redirect based flow visible to the end user and requires the redirect URIs to be configured. So I would recommend removing the customizations related to the backend OAuth2 client apps.

Reply:
@lgp171188 I think there has been a misunderstanding. There was always a separate client for both backend worker and the SSO needed to be set [here](https://github.com/nelc/configuration-secure/blob/master/microsite-config/config.yaml#L95).

This change just simplifies things. We don't need to hard code credentials anymore. Credentials for both SSO and background worker client will be added automatically from the LMS DB. Only a new SSO client will be created. But for the Background worker client, it will use the default one.

Reply:
@shimulch, my point was that we don't need to make the backend worker OAuth2 application name configurable at all (unlike the SSO applications) because we can use the ones automatically created during the deployment. Sorry if that was not clear.

Reply:
@lgp171188 I understand what you are saying. But even though it makes it configurable, by default it uses the one created during deployment. The advantage here is that we don't have to track credentials in the `config.yaml` file at all.

If you think that is not a feature worth having, I can remove it from the change and we can keep tracking credentials in `config.yaml` for background worker client as we do now. :) 

Sorry for keep having this conversation, but in my view, it's a good improvement.

Reply:
@lgp171188 I've removed background worker OAuth2 client from configurable approach. :)